### DISCUSSION TITLE
2-step repeat tests for "real world" application of practice words

### DISCUSSION THREAD

**Discussion by NemukaPrime** [monkeytypegame/monkeytype#2874] [Ideas]
I would like to request a feature regarding repeated tests: a 2-part test that I think could help with effective practice.

Maybe it can be called "Smart Repeat" or something.

**Smart Repeat:** _Practice missed words, and then repeat the previous test where those missed words came from._

I want to be able to repeat the words I missed, and then re-apply that practice to the previous test, so that the missed words practice can be applied in a "real world" context (because in the actual typing tests, we don't just type the same few words repeatedly.)

Example run:

1. The usual typing test with words starts: _show without place interest new have need day stand way_

Mistakes were made with the words "new" and "day"

2. Under "Practice words", select "Toggle smart repeat". And then select "Practice missed" (or slow, or both)
![smart repeat](https://user-images.githubusercontent.com/104338017/165023072-1dd74f26-5ec4-4fa1-b5ff-7ea8fc8ac9b5.png)

 (Mockup of the toggle) 

4. **Smart Repeat Part 1** Typing test starts: _day new new new day day day day new day_

5. After Part 1, **Smart Repeat Part 2** test with the original words list starts: _show without place interest new have need day stand way_

6. Based on user choice, the next test is a new set of words, or Smart Repeat again.

The "Smart Repeat" Part 1 test could also be customized with missed words, slow words, or both.

I apologize if this feature is already built in. Looking forward to hearing your thoughts. Thank you for reading.

**Comment by aNOOBisTheGod** [comment]
**I think this feature can exist, but...**

When you'll finish the test, you can see the button "toggle words history", when you press it, you can see a words, and at the top of the words there is icon with blank paper("copy words list"). You can press it and copy all test words in your clipboard. Then you can practice missed words and after that create same custom test as you had using "paste" function (ctrl + v).

I think the way you suggested will just make website more complex for users.
